py3scd30.py

```diff
@@ -0,0 +1,183 @@
+    eprint("Unknown error: ", sys.exc_type, ":", sys.exc_info()[1])
+
+try:
+	h = pi.i2c_open(I2C_BUS, I2C_SLAVE)
+except:
+	eprint("i2c open failed")
+	exit(1)
+
+# read meas interval (not documented, but works)
+
+def read_n_bytes(n):
+
+  try:
+    (count, data) = pi.i2c_read_device(h, n)
+  except:
+    eprint("error: i2c_read failed")
+    exit(1)
+
+  if count == n:
+    return data
+  else:
+    eprint("error: read measurement interval didnt return " + str(n) + "B")
+    return False
+
+def i2cWrite(data):
+  try:
+    pi.i2c_write_device(h, data)
+  except:
+    eprint("error: i2c_write failed")
+    return -1
+  return True
+
+
+def read_meas_interval():
+  ret = i2cWrite([0x46, 0x00])
+  if ret == -1:
+    return -1
+
+  try:
+    (count, data) = pi.i2c_read_device(h, 3)
+  except:
+    eprint("error: i2c_read failed")
+    exit(1)
+
+  if count == 3:
+    if len(data) == 3:
+      interval = int(data[0])*256 + int(data[1])
+      #print "measurement interval: " + str(interval) + "s, checksum " + str(data[2])
+      return interval
+    else:
+      eprint("error: no array len 3 returned, instead " + str(len(data)) + "type: " + str(type(data)))
+  else:
+    "error: read measurement interval didnt return 3B"
+
+  return -1
+
+read_meas_result = read_meas_interval()
+if read_meas_result == -1:
+  exit(1)
+
+if read_meas_result != 2:
+# if not every 2s, set it
+  eprint("setting interval to 2")
+  ret = i2cWrite([0x46, 0x00, 0x00, 0x02, 0xE3])
+  if ret == -1:
+    exit(1)
+  read_meas_interval()
+
+
+#trigger cont meas
+# TODO read out current pressure value
+pressure_mbar = 972
+LSB = 0xFF & pressure_mbar
+MSB = 0xFF & (pressure_mbar >> 8)
+#print ("MSB: " + hex(MSB) + " LSB: " + hex(LSB))
+#pressure_re = LSB + (MSB * 256)
+#print("press " + str(pressure_re))
+pressure = [MSB, LSB]
+
+pressure_array = ''.join(chr(x) for x in [pressure[0], pressure[1]])
+
+#pressure_array = ''.join(chr(x) for x in [0xBE, 0xEF]) # use for testing crc, should be 0x92
+#print(pressure_array)
+
+f_crc8 = crcmod.mkCrcFun(0x131, 0xFF, False, 0x00)
+
+#print("CRC: " + hex(f_crc8))
+i2cWrite([0x00, 0x10, pressure[0], pressure[1], f_crc8(b'123456789')])
+
+# read ready status
+while True:
+  ret = i2cWrite([0x02, 0x02])
+  if ret == -1:
+    exit(1)
+
+  data = read_n_bytes(3)
+  if data == False:
+    time.sleep(0.1)
+    continue
+
+  if data[1] == 1:
+    #print "data ready"
+    break
+  else:
+    #eprint(".")
+    time.sleep(0.1)
+
+#read measurement
+i2cWrite([0x03, 0x00])
+data = read_n_bytes(18)
+
+#print "CO2: "  + str(data[0]) +" "+ str(data[1]) +" "+ str(data[3]) +" "+ str(data[4])
+
+struct_co2 = struct.pack('>BBBB', data[0], data[1], data[3], data[4])
+float_co2 = struct.unpack('>f', struct_co2)[0]
+
+struct_T = struct.pack('>BBBB', data[6], data[7], data[9], data[10])
+float_T = struct.unpack('>f', struct_T)
+
+struct_rH = struct.pack('>BBBB', data[12], data[13], data[15], data[16])
+float_rH = struct.unpack('>f', struct_rH)[0]
```

Comment:
why did you add [0] for float_co2 and float_rH, but not for float_T - is there any difference?

Reply:
I will test, but I think it was because the co2 and rh returned a list while the temperature returned a single value.

Reply:
Hm - the data format is exactly the same for all three... and struct.unpack('>f', ... ) should return a single float every time...

Reply:
Yes that makes sense, I will have time to test today and will put the output of the code here later.

Reply:
There should definitely be a [0] for temperature as well, you are right :)